numpy/lib/tests/test_mixins.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import division, absolute_import, print_function
+
+import numbers
+
+import numpy as np
+from numpy.testing import TestCase, run_module_suite, assert_, assert_equal
+
+
+class ArrayLike(np.NDArrayOperatorsMixin):
+    """An array-like class that wraps NumPy arrays.
+
+    Example usage:
+        >>> x = ArrayLike([1, 2, 3])
+        >>> x - 1
+        ArrayLike(array([0, 1, 2]))
+        >>> 1 - x
+        ArrayLike(array([ 0, -1, -2]))
+        >>> np.arange(3) - x
+        ArrayLike(array([-1, -1, -1]))
+        >>> x - np.arange(3)
+        ArrayLike(array([1, 1, 1]))
+    """
+
+    def __init__(self, value):
+        self.value = np.asarray(value)
+
+    # __array_priority__ = 1000  # for legacy reasons
+
+    _handled_types = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # ArrayLike implements arithmetic and ufuncs by deferring to the
+        # wrapped array
+        out = kwargs.get('out', ())
+        for x in inputs + out:
+            # handle _handled_types and superclass instances
+            if not (isinstance(x, self._handled_types) or
+                    isinstance(self, type(x))):
+                return NotImplemented
+
+        inputs = tuple(x.value if isinstance(self, type(x)) else x
+                       for x in inputs)
+        if out:
+            kwargs['out'] = tuple(x.value if isinstance(self, type(x)) else x
+                                  for x in out)
+        result = getattr(ufunc, method)(*inputs, **kwargs)
+        if isinstance(result, tuple):
```

Comment:
Needs an early exit path here for `method == 'at'`, else we wrap `None` by accident.

Also, since `ufunc.nout == 1` can't be used for `divmod` here, this check should probably be `type(result) is tuple`.